### DISCUSSION TITLE
Welcome to kagemomiji/airsonic-advanced Discussions!

### DISCUSSION THREAD

**Discussion by kagemomiji** [kagemomiji/airsonic-advanced#102] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to announce important changes and Q&A

**Comment by fault1er1** [comment]
I'm just curious about your future plans with this fork. Since I've got a very large database I think it's not recommended to change the version like two times a week, especially when there were database changes
Would you say it's safe/worth switching from vanilla airsonic-advanced to your fork? I've seen a few changes in vanilla the last 3 days.
Do you intend to maintain the fork in the long term? I'm asking because I've switched like 5 times in the past and I'm afraid to lost all my database entries.

Thank you for continuing the development of airsonic-advanced 👍

**Reply by kagemomiji** [reply]
> I'm just curious about your future plans with this fork. Since I've got a very large database I think it's not recommended to change the version like two times a week, especially when there were database changes

I plan to spend some months or a year mainly working on security issues and fixing some bugs, so I don't plan to make so many breakable changes,

After those works are finished, I have some plans.

1. Treat only new vulnerabilities. 
2. Refactoring codes and implement tests, because the codes of airsonic-advanced are very complicated but not enough test codes to prevent degration.
3. Update UI for support mobile devices without mobile applications (because mobile applications also not been maintained)

> Would you say it's safe/worth switching from vanilla airsonic-advanced to your fork? I've seen a few changes in vanilla the last 3 days.

It's safe if you use my stable version 11.0.2 because it has no changes to db from vanilla. If you want to return to vanilla you can.
But edge version is not safe, because there is DB changes to surpport CUE sheet and it is not stable.

Worthness is difficult to say now, because the changes you mentioned indicates that developers of vanilla airsonic-advanced really restart, you should use vanilla one.

> Do you intend to maintain the fork in the long term? I'm asking because I've switched like 5 times in the past and I'm afraid to lost all my database entries.

Yes I do. I will maintain this fork in the long term, because I want to use airsonic-advanced too.

**Reply by fault1er1** [reply]
Thanks for your kind reply! Soon I'll safe all my db and stuff and will give your fork a try.

**Comment by 8bitbuddhist** [comment]
Thank you for maintaining this! My AA instance failed and became unrecoverable, and I thought I'd have to switch to another tool, but your fork brought it back to life. I'm using the stable branch, and it works great.

**Reply by 8bitbuddhist** [reply]
FYI for anyone else using Docker/Kubernetes/containers: just replace the `airsonicadvanced/airsonic-advanced` image with `ghcr.io/kagemomiji/airsonic-advanced:stable`, restart the container, and you're good to go.

**Comment by ArtiskOnGit** [comment]
Hello, 
First of all, thank you for you work of maintaining airsonic ! I appreciate it  !
I have a question I couldn't find the awnser on the readme, does your fork support last.fm integration ? It is really not important to me, just wanted to know if it works or not and also for others that might be wondering.
Thanks

**Reply by kagemomiji** [reply]
@ArtiskOnGit 
I have tried last.fm integration with my fork, and it works.

Procedure
1. Settings -> Personal ->  check `Register what I'm playing at Last.fm`  -> save
2. Settings -> Credentials ->  click `Add credentials` ->  select `Last.fm` at App and register your username and password of Last.fm

**Reply by ArtiskOnGit** [reply]
I tried that but it doesn't seem to be working. Will try again to see if I made a typo.

**Reply by ArtiskOnGit** [reply]
I entered my last.fm credentials for my admin user that I'm not using to listen to music. Personal settings are for per user settings (lol)

**Comment by vqi418** [comment]
Thank you very much for keeping this alive.
I've been using it for some time, and when the updates stopped in the main repo, I spent quite a while looking for something else to use. Nothing out there that I could find is anywhere near as configurable as AA.
I am very happy to be able to continue using it.
You Rule.
